### PR TITLE
[plugin.video.ytchannels] 0.4.0

### PR DESCRIPTION
**Pull request by mintsoft**
### Description

YouTube Channels, listing and playback 0.4.0

Adding:

-URL encode channel name to support UTF-8 encoding
-Update yt_time function
-Add thumbnail and channel description when searching for new channels to add
-Add option to use Invidious instead of YouTube for playing videos

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki//view/Official:Forum_rules#Piracy_Policy) of this project. 

I believe this to be true, invidious is not a piracy addon, it is a privacy addon and whilst isn't in the main addon repo, it's an optional dependency anyway.

- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.
- If you see no activity on your PR after a week (so at least one weekend has passed) then please go to the #kodi-dev freenode IRC channel to reach out to the team

### CODE REVIEW

File: plugin.video.ytchannels/addon.xml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<addon id="plugin.video.ytchannels" name="YouTube Channels" version="0.3.6+matrix.1" provider-name="natko1412,mintsoft,yeahme49">
+<addon id="plugin.video.ytchannels" name="YouTube Channels" version="0.4.0+matrix.1" provider-name="natko1412,mintsoft,yeahme49">
 	<requires>
 		<import addon="xbmc.python" version="3.0.0"/>
 		<import addon="script.module.six" version="1.11.0"/>
 		<import addon="plugin.video.youtube"/>
+		<import addon="plugin.video.invidious" optional="true"/>
 	</requires>
```
Comment (enen92) [inline plugin.video.ytchannels/addon.xml]: What is plugin invidious? We don't have this on our repo..

Comment (mintsoft) [reply plugin.video.ytchannels/addon.xml]: It is this: https://github.com/TheAssassin/kodi-invidious-plugin which uses https://github.com/iv-org/invidious under the hood.

It's essentially a privacy proxy for Youtube, it's an optional feature and the settings are only visible if the plugin is enabled already.

**Comment by mintsoft** [comment]
@enen92 any update on this? I did check the rules (https://kodi.wiki/view/Add-on_rules) before merging this change into the plugin and there's nothing in there about not having optional dependencies on non-kodi-repo-addons?

**Comment by enen92** [comment]
Sorry for the delay. I brought this issue to the team and the general consensus was to disallow such behaviour. While the plugin today is innofensive we have no control over it so theres no guarantee it will stay like this in the future. Why not ask the author to submit the add-on to the repo?

**Comment by mintsoft** [comment]
Yeah fair enough @enen92 I'll see if we can have that conversation

**Comment by mintsoft** [comment]
@enen92 I've removed invidious support and dependency from this version as it doesn't look like the invidious guys have much interest/time to get it in the main repo :)

**Review by enen92 — APPROVED** [review]
Thanks

**Comment by petterreinholdtsen** [comment]
Note, since https://github.com/xbmc/repo-plugins/pull/3927 the invidious plugin is in matrix, and since https://github.com/xbmc/repo-plugins/pull/4363 it is in nexus.

**Pull request merged**